src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -3233,6 +3302,34 @@ boolean using(Context context, DeclarationList declList) throws ParserException
         return true;
     }
 
+    private String explicitCast(Type from, Type to, boolean override) {
+        String ecmn = explicitCastMethodName(to.javaName);
+        String res = "    ";
+        if (override) res += "@Override ";
+        res += "public " + to.javaName + ' ' + ecmn + "() { return " + ecmn + "(this); }\n";
+        /* We generate the adapter-aware version of the method when the target class has any annotation,
+           assuming that this annotation is @SharedPtr or some other adapter storing a share_ptr in owner.
+           This is the only use case for now. We can generalize to any adapter if the need arises and call
+           some cast function on the adapter instead of static_pointer_cast. */
+        String annotation = getConstructorAdapter(to);
+        if (annotation != null)
+            res += "    @Namespace public static native " + annotation + " @Name(\"SHARED_PTR_NAMESPACE::static_pointer_cast<" + to.cppName + ">\") "
+                + to.javaName + " " + ecmn + "(@Cast({\"\", \"SHARED_PTR_NAMESPACE::shared_ptr<" + from.cppName + ">\"}) " + annotation + " " + from.javaName + " pointer);\n";
```

Comment:
If you need a cast here for this to work, put it on the constructor in your presets, not here.

Reply:
I replaced the `@Cast` by making explicit of the second template parameter in `static_pointer_cast`. Commit pushed.